src/move_basic.cpp

```diff
@@ -407,11 +410,19 @@ void MoveBasic::executeAction(const move_base_msgs::MoveBaseGoalConstPtr& msg)
     std::string planningFrame;
     double goalYaw;
 
-    tf2::Transform goalPlanning;
-    if (!transformPose(frameId, preferredPlanningFrame, goal, goalPlanning)) {
-        ROS_WARN("MoveBasic: Will attempt to plan in %s frame", frameId.c_str());
+    // The pose of the robot planning frame MUST be known initially, and may or may not
+    // be known after that.
+    // The pose of the robot in the driving frame MUST be known at all times.
+    // A planning frame of "goal" means to use what ever frame the goal is specified in.
+    if (preferredPlanningFrame == "goal") {
```

Comment:
Maybe instead of "goal" this behavior should be when the `preferredPlanningFrame` is empty string?

That way if someone decides name a frame "goal" we don't have a weird problem. also this seems like the behavior you would want if you don't have a preferred frame.

Reply:
Update so the empty string means use the frame specified in the goal.